### ISSUE TITLE
Default-named functions can't be found with `BinaryView.get_symbols_by_name`

### ISSUE THREAD

**Issue by nbsdx** [Vector35/binaryninja-api#511]
Repro:

1. Open a function that's auto-named by Binary Ninja that's of the format `sub_XXXXXX`. 
2. In the python shell: `bv.get_symbols_by_name(current_function.name)`
3. Result is `[]`

**Comment by nbsdx** [comment]
Additionally, these symbols are not returned with a call to `BinaryView.get_symbols()`

**Comment by psifertex** [comment]
That's because those aren't actually symbols. They're what are rendered if you don't have a symbol. You can see that by creating a symbol then deleting it. It will revert back to that auto-name. 

That's currently by design -- is there a reason that doesn't work for something you're working on?

**Comment by nbsdx** [comment]
That makes sense. 

The replacement function view that I made currently relies on `BinaryView.get_symbols_by_name` to find the address of the function to navigate to. Looking around some more, I could use `BinaryView.functions`, and check each name individually, which shouldn't be too bad unless there are a ton of functions.

**Comment by nbsdx** [comment]
I'm going to mark this as closed since the behavior is intended. 

`syms = [x for x in bv.functions if x.name == TARGET]` works just fine for what I'm doing.

**Comment by psifertex** [comment]
Sorry for the delay, dodging hurricanes. ;-)

I'm still a bit confused as to the original use case. If you've already got `current_function.name`, what do you need to lookup the function for that name? You've already got `current_function`...

**Comment by nbsdx** [comment]
No problem :)

So, `current_function.name` is a bit of a simplification of what I was trying to do. In my Function List UI modification, I replace the `QListView` with my own `QTableView` and install a `QSortFilterProxyModel` in between the existing model on the `QListView`, and my new `QTableView` (I wanted to be able to sort functions by name on top of the filtering).

The original Model (which I'm piggybacking on), only contains the text that corresponds to the `name` field in the Python `Function` class (I feel like this is a poor explanation, but I'm not sure how else to word it...). That `name` field is generated from some `Symbol` stuff that I didn't feel like diving through, but from what you said, and what I saw, it creates a temporary `Symbol` on the fly that's thrown away afterwards. 

Since I don't have access to the internal Binary Ninja functions that navigate to a function based on the model index that's selected, I'm stuck looking up the function by name using the current `BinaryView`. I could probably instead look at `"sub_FUNCNAME".split("_")[0]`, but that looks super hacky. 

Now, while typing this up, I realized that I could likely be able to invoke the `goToFunction(QModelIndex)` slot through the Meta Object system, so I'm going to look at doing that instead :) (Maybe, I've been having issues with signals & slots from PyQt)
